counter.go

```diff
@@ -16,11 +16,12 @@ package nftables
 
 import (
 	"github.com/google/nftables/binaryutil"
+	"github.com/google/nftables/expr"
 	"github.com/mdlayher/netlink"
 	"golang.org/x/sys/unix"
 )
 
-// CounterObj implements Obj.
+// Deprecated: Use ObjAttr instead
```

Comment:
Instead of deprecating these types, would it be possible to convert to ObjAttr under the covers? Maybe I’m not fully understanding the before/after, though. Would you mind showing an example change that users of CounterObj need to do?

Reply:
I've already changed CounterObj and QuoteObj to implement Obj interface as a form of compatibility with functions in `obj.go`. With this change, the only thing that users of CounterObj should change is the conversion of the returned object that implements the Obj interface.

Here are a few examples:
```go
// This will work since CounterObj implements Obj
obj, err := c.ResetObject(&nftables.CounterObj{
	Table: filter,
	Name:  "fwded",
})

// This will work as well
counter1 := c.AddObj(&nftables.CounterObj{
	Table:   table,
	Name:    "fwded1",
	Bytes:   1,
	Packets: 1,
})
// Also works, although counter1 is technically ObjAttr
obj1, err := c.GetObject(counter1)
// This wouldn't work anymore, obj1 is ObjAttr
rcounter1, ok := obj1.(*nftables.CounterObj)
// Previous line should be changed to this
rcounter1, ok := obj1.(*nftables.ObjAttr)
```

We could add a special case when we unmarshal ObjTypeCounter and ObjTypeQuota to return CounterObj and QuotaObj instead of ObjAttr, but this approach kind of beats the purpose of using ObjAttr for these two types since we would always leverage legacy types instead of new ObjAttr.

Let me know how to approach this.

Reply:
Hi @stapelberg, 

this discussion has gone stale a bit. I have refactored this feature to make it backwards compatible. The implementation now detects when user passes the "new" `ObjAttr` type and then works with and returns the `ObjAttr` value, otherwise we default to "old" `QuotaObj` or `CounterObj`. I have implemented tests using both variants to demonstrate how it works.

Let me know what you think.